woocommerce/woocommerce
# Enable cart item images filtering, fixes #42259

### Changes proposed in this Pull Request:

Adding a filter hook named `woocommerce_cart_item_images` that receives 3 parameters:

 - `$product_images` An array of image objects to be filtered.
 - `$cart_item` The cart item the images belong to.
 - `$cart_item_key` The cart item key.

This way a plugin can edit cart item images on a per item basis.



### How to test the changes in this Pull Request:

Code review is probably sufficient for this change. However, you can easiliy test this change by running through the following steps:

1. Install WooCommerce
2. Activate WooCommerce
3. Add a product
4. _Optional: Add images to the product_
5. Install and activate a theme with cart-blocks support, for example storefront
6. Add the created product to the cart. You should now see either the selected images, or a placeholder in the cart
7. Create a test plugin using the following code:
```
<?php

/**
 * @wordpress-plugin
 * Plugin Name:       ReplaceCartThumbnail
 * Description:       This plugin replaces cart thumbnails with random images
 */

// Discard the original product images array, and returns a new array with a replaced image
add_filter( 'woocommerce_cart_item_images', function ($productimages, $cart_item, $cart_item_key)
{
  $image_path = "https://picsum.photos/seed/$cart_item_key/200";
  return [
    (object)[
      'id'        => (int) 0,
      'src'       => $image_path,
      'thumbnail' => $image_path,
      'srcset'    => (string)'',
      'sizes'     => (string)'',
      'name'      => 'Random product image',
      'alt'       => 'Random product image',
    ]
  ];
}, 10, 3 );
```

8. Install the test plugin and activate
9. Go back to the cart.
10. The cart should now show a random image as the thumbnail

### Changelog entry




<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Enables replacing cart item images in cart-blocks using a filter. Complementary shortcode hook already exists: woocommerce_cart_item_thumbnail.

</details>
